core/src/main/java/nu/yona/server/subscriptions/service/UserUpdateService.java

```diff
@@ -115,21 +116,44 @@ public UserDto updateUser(UUID id, UserDto user)
 			assertValidUpdateRequestForUserCreatedNormally(user, userEntity);
 			UserDto originalUser = userLookupService.createUserDto(userEntity);
 
-			user.updateUser(userEntity);
-			Optional<ConfirmationCode> confirmationCode = createConfirmationCodeIfNumberUpdated(user.getMobileNumber(),
-					originalUser.getMobileNumber(), userEntity);
-			if (confirmationCode.isPresent())
-			{
-				sendConfirmationCodeTextMessage(userEntity.getMobileNumber(), confirmationCode.get(),
-						SmsTemplate.CHANGED_USER_NUMBER_CONFIRMATION);
-			}
+			updateUser(user, userEntity, originalUser);
+			sendConfirmationCodeIfNumberUpdated(user, userEntity, originalUser);
+
 			UserDto userDto = userLookupService.createUserDto(userEntity);
 			logger.info("Updated user with mobile number '{}' and ID '{}'", userDto.getMobileNumber(), userDto.getId());
 			buddyService.broadcastUserInfoChangeToBuddies(userEntity, originalUser);
 		});
 		return userLookupService.createUserDto(updatedUserEntity);
 	}
 
+	private void sendConfirmationCodeIfNumberUpdated(UserDto user, User userEntity, UserDto originalUser)
+	{
+		Optional<ConfirmationCode> confirmationCode = createConfirmationCodeIfNumberUpdated(user.getMobileNumber(),
+				originalUser.getMobileNumber(), userEntity);
+		if (confirmationCode.isPresent())
+		{
+			sendConfirmationCodeTextMessage(userEntity.getMobileNumber(), confirmationCode.get(),
+					SmsTemplate.CHANGED_USER_NUMBER_CONFIRMATION);
+		}
+	}
+
+	private void updateUser(UserDto user, User userEntity, UserDto originalUser)
+	{
+		if (user.getCreationTime().isPresent() && !user.isChanged(originalUser))
```

Comment:
This is a quite strange condition. If the user is changed, it is suddenly allowed to send a new creation date?

Reply:
You're overlooking the exclamation mark in front of ``user.isChanged``. If the creation date is present and the DTO does *not* contain a property update, then it is accepted as a creation date change.